### PR TITLE
Replace origin proof timestamp with nonce

### PR DESCRIPTION
**Pull request by yuval-netanel**

### CODE REVIEW

**Comment by grozen** [comment]
Highly approved.

File: KlarnaOnDemandTests/KODOriginProofSpec.m [hunk 1]
```diff
@@ -35,17 +28,29 @@ + (NSString *)timestamp;
     [[dataDic[@"amount"] should] equal:[NSNumber numberWithInt:3600]];
     [[dataDic[@"currency"] should] equal:@"SEK"];
     [[dataDic[@"user_token"] should] equal:@"my_token"];
-    [[dataDic[@"timestamp"] should] equal:[KODOriginProof timestamp]];
+    [[dataDic[@"id"] should] matchPattern:@"[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"];
```
Comment (tomerlevi444) [inline KlarnaOnDemandTests/KODOriginProofSpec.m]: Create a variable for this string. It's more self-explanatory.

File: KlarnaOnDemandTests/KODOriginProofSpec.m [hunk 2]
```diff
@@ -35,17 +28,29 @@ + (NSString *)timestamp;
...
+    NSString *originProofB = [KODOriginProof generateWithAmount:3600 currency:@"SEK" userToken:@"my_token"];
+    NSData *decodedriginProofB = [[NSData alloc] initWithBase64EncodedString:originProofB options:0];
+    NSDictionary *originProofDicB = [NSJSONSerialization JSONObjectWithData: decodedriginProofB
+                                                                    options: NSJSONReadingMutableContainers
+                                                                      error: nil];
+    NSData *dataB = [originProofDicB[@"data"] dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *dataDicB = [NSJSONSerialization JSONObjectWithData:dataB options:0 error:nil];
+    
+    [[dataDicA[@"id"] shouldNot] equal:dataDicB[@"id"]];
+  });
```
Comment (tomerlevi444) [inline KlarnaOnDemandTests/KODOriginProofSpec.m]: You can create a helper method that gets the internal decoded data from the origin proof.

**Comment by tomerlevi444** [comment]
Approved!

**Pull request merged**